central/reports/scheduler/v2/scheduler.go

```diff
@@ -26,7 +24,7 @@ type Scheduler interface {
 
 	// CancelReportRequest cancels a report request that is still waiting in queue.
 	// If the report is already being prepared or has completed execution, it cannot be cancelled.
-	CancelReportRequest(ctx context.Context, reportID string) (bool, error)
+	CancelReportRequest(reportID string) (bool, error)
```

Comment:
This should accept caller's context. This is not called async.

Reply:
It could. But should it use the user context for deleting the report job from db? Same for SubmitReportRequest() too, it uses the `allAccessContext` to add the snapshot with WAITING state to db.

Reply:
The rule of thumb is that functions that interact and manipulate data must accept a context. Generally, the lower components will not change as fast as higher-level components. Thus, if the contract at the higher level changes, we mistakenly do not give privileged access.

Reply:
Nevermind, since deleting from snapshot datastore should require `WorkflowAdministration` write permission too, it makes sense to use the user context for deleting as well.